process/sync/baseForkDetector.go

```diff
@@ -457,16 +482,55 @@ func (bfd *baseForkDetector) shouldAddBlockInForkDetector(
 	finality int64,
 ) error {
 
-	noncesDifference := int64(bfd.ProbableHighestNonce()) - int64(header.GetNonce())
-	isSyncing := state == process.BHReceived && noncesDifference > process.MaxNoncesDifference
-	if state == process.BHProcessed || isSyncing {
+	if state == process.BHProcessed || bfd.isSyncing(header.GetNonce()) {
 		return nil
 	}
 
+	// This check is very important for meta-chain, as it prevents the situation when whole shard would been reverted
```

Comment:
rewrite the comment. Make short sentences. You might describe the problem with bulletpoints. These comments could be added in the already existing fork choice document. no need to keep here. With this code we are back to the initial issue, where metachain remained stucked at nonce and this condition does not seam right for shard chains.

Reply:
Actually, this condition is very important to avoid some stuck situation, when shards notarized as final block n because they received n+1 but the metachain will revert to an older n because they received this older n with latency but before n+1 and in this case they all reverted to the first n. 
The problem with initial issue, when a network could be split for whatever reason, has been solved by introducing this mechanism of forced forks which is triggered in a modulus round and only if after some certain rounds the consensus could not advance. Actually all the nodes will rollback to their same view which is the final block, their probable higher nonce will be reset to this block and they would entered in the consensus and would proposed a new block with nonce final + 1.

Reply:
Comment has been rewritten